commit 0b900fd3581f828b8b502667450a200322d24c1f (refs/changes/45/4945/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-10-12T00:39:24+00:00 (12 months ago)
    
    joyent/node-sshpk#52 Buffer no longer performs length check for hex strings in node v8.x (fix silly typo)

diff --git a/lib/fingerprint.js b/lib/fingerprint.js
index 62f3754..3ddd62a 100644
--- a/lib/fingerprint.js
+++ b/lib/fingerprint.js
@@ -102,7 +102,7 @@ Fingerprint.parse = function (fp, options) {
 		parts = parts.join('');
 		/*JSSTYLED*/
 		var md5RE = /^[a-fA-F0-9]+$/;
-		if (!md5RE.test(parts) || md5RE.length % 2 !== 0)
+		if (!md5RE.test(parts) || parts.length % 2 !== 0)
 			throw (new FingerprintFormatError(fp));
 		try {
 			hash = Buffer.from(parts, 'hex');
diff --git a/test/fingerprint.js b/test/fingerprint.js
index a74b5ac..6d04273 100644
--- a/test/fingerprint.js
+++ b/test/fingerprint.js
@@ -84,6 +84,18 @@ test('fingerprint matches', function(t) {
 	    'SHA256:PYC9kPVC6J873CSIbfp0LwYeczP/W4ffObNCuDJ1u5w');
 	t.ok(f.matches(k2));
 	t.ok(!f.matches(k1));
+	var f2 = sshpk.parseFingerprint(
+	    '59:a4:61:0e:38:18:9f:0f:28:58:2a:27:f7:65:c5:87');
+	t.ok(f2.matches(k1));
+	t.ok(!f2.matches(k2));
+	var f3 = sshpk.parseFingerprint(
+	    'MD5:59:a4:61:0e:38:18:9f:0f:28:58:2a:27:f7:65:c5:87');
+	t.ok(f3.matches(k1));
+	t.ok(!f3.matches(k2));
+	var f4 = sshpk.parseFingerprint(
+	    'SHA1:3JP2y/wCv8KnvAunLz7EjcEhKeE');
+	t.ok(f4.matches(k1));
+	t.ok(!f4.matches(k2));
 	t.end();
 });
 
